dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/AbstractServiceDiscovery.java

```diff
@@ -150,7 +150,7 @@ public synchronized void register() throws RuntimeException {
         }
         if (this.serviceInstance == null) {
             ServiceInstance serviceInstance = createServiceInstance(this.metadataInfo);
-            if (!isValidInstance(serviceInstance)) {
+            if (!isValidInstance(serviceInstance, applicationModel)) {
```

Comment:
This might cause an invalid instance to be registered to the registry for the first time and never get updated.

Reply:
ok, Change the new defaultServiceInstance to createServiceInstance